sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/Aggregator.java

```diff
@@ -74,6 +74,15 @@ default T accumulateDoubleMeasurement(double value, Attributes attributes, Conte
     return handle.accumulateThenReset(attributes);
   }
 
+  /**
+   * Validate that the asynchronous {@code currentAccumulation} is compatible with the {@code
+   * previousAccumulation}.
+   *
+   * @throws Exception if the {@code currentAccumulation} is invalid
+   */
+  default void validateAsyncAccumulation(T previousAccumulation, T currentAccumulation)
```

Comment:
The only aggregator that implements this are the sum aggregators `DoubleSumAggregator` and `LongSumAggregator`, which is a smell. 

This might be more generally useful as a reset detection mechanism. I.e. during collection we ask the aggregator if the current accumulation is compatible with the previous accumulation, and if not, we indicate that a reset a reset in the `MetricData`.

Reply:
You'd also want to do this for Histogram if we ever support gauge/async histograms.  The "sum" and "count" in histogram need to match w/ Prometheus sum/count sub-metrics for compatibility.